### DISCUSSION TITLE
Mac OS Macbook pro- Desktop Black screen

### DISCUSSION THREAD

**Discussion by timcanty** [Ylianst/MeshCentral#3665] [Q&A]
I didn't want to open this as an issue straight away, as not sure if it's an issue our side or not. We have only tried this on one mac so far, as we are mainly a windows company. Apologies if this is the wrong place to post it.

So we have installed the agent on the mac book, it allows us to use the terminal and other features, however when we try connect to the desktop, all it gives us is a black screen.

The macbook is quite a fresh install, it is running Mac OS Monterey, i haven't got it in front of me to get the exact version currently, but i can get hold of this early next week if required. We manually added the agent into the privacy->, accessibility and screen recording as there wasn't a pop up on first login, but still whenever we click the connect button we just get a black screen.

 Not sure if this could be firewall related, but couldn't see anything been blocked, something odd in our environment, or a possible bug.

Would like to help try find out if this is a problem our side, or a bug, but we are quite new to using mesh central so not sure if there is any debug options that we can turn on that may assist in getting any information to point in the direction of why its not working,

We have limited time to play with the macbook, but we can arrange early next week to have it for a short while to try any debug options.

**Comment by JSuenram** [comment]
Had same problem here Intel MacBook Pro,
Just remove all Services etc which are Installed from MeshCentral-Agent and start from scratch.

Did work here. So I can say the MacOS-Agent is working as expected, but sometimes MacOS-Security-Settings get weird...

**Comment by krayon007** [comment]
I would suggest you also remove all the security settings you added for the agent, so that the agent can re-request them. I know macOS can be silly sometimes, where if it sees a binary with the same name/path in the allowed list, it will not give you a prompt, even tho it thinks it doesn't have permissions. (I have seen this issue when the hash of the binary is different, but the filename is the same). I have filed a bug against Apple for this specific behavior but so far it has been ignored.

The security popup should have occurred the first time you tried to connect the desktop.
